src/main/java/net/dv8tion/jda/api/managers/GuildScheduledEventManager.java

```diff
@@ -0,0 +1,299 @@
+     * and will automatically be cancelled a few hours after the start time if not.
+     *
+     * @param startTime The time that the selected {@link GuildScheduledEvent} is set to
+     *                  start at.
+     *
+     * @return GuildScheduledEventManager for chaining convenience
+     *
+     * @throws java.lang.IllegalArgumentException If the provided start time is {@code null}, or takes place after the
+     * end time.
+     *
+     * @see #setEndTime(OffsetDateTime)
+     */
+    GuildScheduledEventManager setStartTime(@Nonnull OffsetDateTime startTime);
+
+    /**
+     * Sets the time that the selected {@link GuildScheduledEvent} should end at.
+     * Events of {@link GuildScheduledEvent.Type#EXTERNAL Type.EXTERNAL} will automatically
+     * end at this time, and events of {@link GuildScheduledEvent.Type#STAGE_INSTANCE Type.STAGE_INSTANCE}
+     * and {@link GuildScheduledEvent.Type#VOICE Type.VOICE} will end a few minutes after the last
+     * person has left the channel.
+     *
+     * @param endTime
+     *        The time that the selected {@link GuildScheduledEvent} is set to end at,
+     *        or {@code null} for no end time to be set.
+     *
+     * @return GuildScheduledEventManager for chaining convenience
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided end time is before the start time
+     *
+     * @throws java.lang.IllegalStateException
+     *         If the provided end time is {@code null} when the event is set to take place at an external location
+     *
+     *
+     * @see #setStartTime(OffsetDateTime)
+     */
+    GuildScheduledEventManager setEndTime(@Nullable OffsetDateTime endTime);
+
+    /**
+     * Sets the status of the event. This method may be used to start, end or cancel an event but can only be used to
+     * complete one of the following transitions:
+     * <ol>
+     *     <li>{@link GuildScheduledEvent.Status#SCHEDULED Status.SCHEDULED} --> {@link GuildScheduledEvent.Status#SCHEDULED Status.ACTIVE}</li>
+     *     <li>{@link GuildScheduledEvent.Status#SCHEDULED Status.SCHEDULED} --> {@link GuildScheduledEvent.Status#CANCELED Status.CANCELED}</li>
+     *     <li>{@link GuildScheduledEvent.Status#ACTIVE Status.ACTIVE} &nbsp; &nbsp; &nbsp; --> {@link GuildScheduledEvent.Status#COMPLETED Status.COMPLETED}</li>
+     * </ol>
+     *
+     * @param status
+     *        The new status
+     *
+     * @return GuildScheduledEventManager for chaining convenience
+     *
+     * @throws java.lang.IllegalStateException
+     *         If the transition between statuses does not follow one of the three documented above.
+     * @throws IllegalArgumentException
+     *         If the provided status is {@code null}
+     *
+     * @see #getGuildScheduledEvent()
+     * @see GuildScheduledEvent#getStatus()
+     */
+    GuildScheduledEventManager setStatus(@Nonnull GuildScheduledEvent.Status status);
```

Comment:
All these methods need to have the annotations  `@Nonnull` and `@CheckReturnValue`